{
  try {
  }
 catch (  RuntimeException g) {
    try {
      foo();
    }
 catch (    FileNotFoundException e) {
      throw e;
    }
catch (    IOException e) {
      throw g;
    }
  }
}
